src/Uplink/Admin/Fields/Field.php

```diff
@@ -0,0 +1,312 @@
+<?php declare( strict_types=1 );
+
+namespace StellarWP\Uplink\Admin\Fields;
+
+use StellarWP\Uplink\Admin\License_Field;
+use StellarWP\Uplink\Uplink;
+use StellarWP\Uplink\Config;
+use StellarWP\Uplink\Resources\Collection;
+use StellarWP\Uplink\Resources\Resource;
+// Use function statement is problematic with Strauss.
+use StellarWP\Uplink as UplinkNamespace;
+
+class Field {
```

Comment:
Would this at all work as  a [Controller Component ](https://github.com/stellarwp/uplink/blob/1276bda128b086deeb318dc2eab387fa606b434f/src/Uplink/Components/Controller.php) instead?

Like the [Authorize Button Controller](https://github.com/stellarwp/uplink/blob/1276bda128b086deeb318dc2eab387fa606b434f/src/Uplink/Components/Admin/Authorize_Button_Controller.php#L64) it could take what it needs as an array of arguments, and any dependencies can actually be injected into the constructor instead.

You could also add a function to functions.php e.g. `get_license_field()` so that's the interface people could use.

Note I did an implementation over here that **_I really don't like at all_**: https://github.com/stellarwp/uplink/blob/d37e2bb322e26063bae99fb378f89dab84a6c862/src/Uplink/Admin/License_Field.php but might be worth looking at.